### ISSUE TITLE
register to path automatically

### ISSUE THREAD

**Issue by jinfagang** [lapce/lapce#1082]
one of biggest advantage of vscode is it can summon out wherever you are with `code`, does lapse can also with some like that?

**Comment by panekj** [comment]
you can run `lapce` in terminal if it's path is in your `PATH` environment variable

**Comment by jinfagang** [comment]
I installed on Mac, which directly drop a app to Application folder, this shouldn't be in PATH default. But vscode has not problem not this.

**Comment by cicorias** [comment]
what vscode does is makes the add/remove of this easy via it's command pallete. 

ultimately, it drops the following in `/usr/local/bin` with a file name of `code`

```
#!/usr/bin/env bash
#
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

# when run in remote terminal, use the remote cli
if [ -n "$VSCODE_IPC_HOOK_CLI" ]; then
	REMOTE_CLI="$(which -a 'code' | grep /remote-cli/)"
	if [ -n "$REMOTE_CLI" ]; then
		"$REMOTE_CLI" "$@"
		exit $?
	fi
fi

function app_realpath() {
	SOURCE=$1
	while [ -h "$SOURCE" ]; do
		DIR=$(dirname "$SOURCE")
		SOURCE=$(readlink "$SOURCE")
		[[ $SOURCE != /* ]] && SOURCE=$DIR/$SOURCE
	done
	SOURCE_DIR="$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )"
	echo "${SOURCE_DIR%%${SOURCE_DIR#*.app}}"
}

APP_PATH="$(app_realpath "${BASH_SOURCE[0]}")"
if [ -z "$APP_PATH" ]; then
	echo "Unable to determine app path from symlink : ${BASH_SOURCE[0]}"
	exit 1
fi
CONTENTS="$APP_PATH/Contents"
ELECTRON="$CONTENTS/MacOS/Electron"
CLI="$CONTENTS/Resources/app/out/cli.js"
ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" --ms-enable-electron-run-as-node "$@"
exit $?
```

**Comment by frob** [comment]
This is also an issue if installed with Homebrew. It doesn't seem to differentiate between installing a cask (gui app) or cli.

**Comment by shinspiegel** [comment]
I would like to help with this feature,   
I'm far from an expert in Rust, but with some guidance, I would love to help.  

How should I proceed?
Thanks in advance

**Comment by panekj** [comment]
Fixed via #1471
